src/main.c

```diff
@@ -340,27 +344,65 @@ do_shutdown (GObject  *object,
     notify_uninit();
 }
 
+static int global_retval = 0;
+
+void
+sighandler(int v)
+{
+    global_retval = 128 + v; // exit value is usually 128 + signal_id
+}
+
 int
 main(int argc, char **argv)
 {
+    int retval;
+    int pid = fork();
+    if (pid != 0) {
+        if (pid > 0) {
+            // SIGTERM and SIGINT will be ignored, but
+            // will break WAITPID with a EINTR value in
+            // errno.
+            // This allows the program to always exit with
+            // a NO-ERROR value, and kill the child
+            struct sigaction signal_data;
+            signal_data.sa_handler = sighandler;
+            sigemptyset(&signal_data.sa_mask);
+            signal_data.sa_flags = 0;
+            sigaction(SIGTERM, &signal_data, NULL);
+            sigaction(SIGINT, &signal_data, NULL);
+
+            retval = waitpid(pid, NULL, 0);
+            if ((retval != 0) && (errno == EINTR)) {
```

Comment:
Should this be `retval < 0`? The waitpid call shouldn't ever return 0, since we haven't passed WNOHANG. If the child has exited, then the call will return `pid`.